backend/api/v1/sql_service.go

```diff
@@ -1364,20 +1386,67 @@ func (s *SQLService) getSensitiveSchemaInfo(ctx context.Context, instance *store
 			return nil, errors.Errorf("database %q not found", databaseName)
 		}
 
-		policyResourceType := api.PolicyResourceTypeDatabase
-		policyType := api.PolicyTypeMasking
-		policy, err := s.store.GetPolicyV2(ctx, &store.FindPolicyMessage{
-			ResourceType: &policyResourceType,
-			Type:         &policyType,
-			ResourceUID:  &database.UID,
-		})
+		var project *store.ProjectMessage
+		if database.ProjectID != "" {
+			project, err = s.store.GetProjectV2(ctx, &store.FindProjectMessage{
+				ResourceID: &database.ProjectID,
+			})
+			if err != nil {
+				return nil, errors.Wrapf(err, "failed to find project %q", database.ProjectID)
+			}
+		}
+
+		var maskingExceptionPolicy *storepb.MaskingExceptionPolicy
+		// If we cannot find the maskingExceptionPolicy before, we need to find it from the database and record it in cache.
+		if _, ok := maskingExceptionPolicyMap[database.ProjectID]; !ok {
+			if project != nil {
+				policy, err := s.store.GetMaskingExceptionPolicyByProjectUID(ctx, project.UID)
+				if err != nil {
+					return nil, errors.Wrapf(err, "failed to find masking exception policy for project %q", project.ResourceID)
+				}
+				// It is safe if policy is nil.
+				maskingExceptionPolicyMap[database.ProjectID] = policy
+			}
+		}
+		maskingExceptionPolicy = maskingExceptionPolicyMap[database.ProjectID]
+
+		// Build the filtered maskingExceptionPolicy for current principal.
+		var maskingExceptionContainsCurrentPrincipal []*storepb.MaskingExceptionPolicy_MaskingException
+		if maskingExceptionPolicy != nil {
+			for _, maskingExceptionPolicy := range maskingExceptionPolicy.MaskingExceptions {
+				for _, member := range maskingExceptionPolicy.Members {
+					if member == currentPrincipal.Email {
+						maskingExceptionContainsCurrentPrincipal = append(maskingExceptionContainsCurrentPrincipal, maskingExceptionPolicy)
+						break
+					}
+				}
+			}
+		}
+
+		// Filtered the current project's data classification config.
+		var dataClassificationConfig *storepb.DataClassificationSetting_DataClassificationConfig
+		if project.DataClassificationConfigID != "" {
```

Comment:
project could be nil

Reply:
NVM, each database must belong to a database